### ISSUE TITLE
Support varargs

### ISSUE THREAD

**Issue by edsko** [well-typed/hs-bindgen#53]
It's not really clear _how_ to support varargs exactly. The `CAPI` convention _does_ have support varargs functions, but it's [very limited](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/ffi.html#varargs-not-supported-by-ccall-calling-convention): you have to specify a precise set of arguments, and `ghc` then generates a wrapper that takes precisely those arguments and in turn calls the original varargs function. 

https://stackoverflow.com/questions/5989457/haskell-ffi-support-for-functions-with-variadic-arguments lists some options here; the most promising is perhaps [using `libffi` to push an arbitrary number of arguments to the stack](https://stackoverflow.com/a/5993358/742991) before calling the function (though obviously if we do go down this road we should provide a nice and type-safe API for this).

We will need examples of what exactly is desirable before we can make progress here.

**Comment by edsko** [comment]
Closing this as `wontfix`. As of #743 we _check_ (and skip generating bindings) for variadic functions. While it is possible to provide a type-unsafe API as per the StackOverflow answer linked above, this could be done independently from `hs-bindgen`. Providing a type-safe API is impossible; the C signature does not give us enough information to do so.
